### DISCUSSION TITLE
Debugging a core dump

### DISCUSSION THREAD

**Discussion by sei-jgwohlbier** [iree-org/iree#17462] [Runtime]
Hi,
I have successfully run the `simple_abs` and `mnist` examples on a gpu, but my own model dumps core. Any hints on how to debug this?

Also, I'm not sure how to specify `--input` for my network as it has two input tensors. That might be my next issue, but I don't think the runtime is getting that far at the moment. My function signature is `func.func @forward(%arg0: tensor<1x2xf32>, %arg1: tensor<1x1xf32>) -> tensor<1xf32>`

```
iree-run-module \
--device=cuda \
--module=mymodel.vmfb \
--function=forward \
--input="1x2xf32,1x1xf32=0.5,0.4"
Aborted (core dumped)
make: *** [Makefile:37: run] Error 134
```

**Comment by ScottTodd** [comment]
Each input should be specified separately, like `--input=1x2xf32 --input=1x1xf32=0.5,0.4`. See https://iree.dev/developers/general/developer-overview/#iree-run-module for more examples

For core dumps, you'll want to build with debug symbols then consider running under a debugger. Either of those should get you useful error messages and stack traces.

**Reply by sei-jgwohlbier** [reply]
Thanks! That indeed fixed the core dump and the model appears to execute correctly!

I'm planning to use iree to generate ptx and link that ptx to another application that uses `nvcc` and `clang` to generate GPU code. If you have any thoughts about that I'd love to hear them. I'm still mulling over how I'll make the function call from the host code to get the proper signature for linking to the iree generated ptx.

Thanks!
